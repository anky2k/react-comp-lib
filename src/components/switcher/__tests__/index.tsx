import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Switcher from '../../switcher';
import UiProvider from '../../ui-provider';

afterEach(cleanup);

test('it should render as per given styles', () => {
  const { asFragment } = render(<Switcher />);

  expect(asFragment()).toMatchSnapshot();
});

test('it should toggle to a different theme mode', () => {
  const { getByTestId, getByTitle } = render(
    <UiProvider>
      <Switcher />
    </UiProvider>
  );
  fireEvent.click(getByTestId('astra-theme-switch'));
  expect(getByTitle('switch to light mode')).toBeInTheDocument();
  fireEvent.click(getByTestId('astra-theme-switch'));
  expect(getByTitle('switch to dark mode')).toBeInTheDocument();
});
