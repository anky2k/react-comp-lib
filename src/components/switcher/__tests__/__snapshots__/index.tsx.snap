// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render as per given styles 1`] = `
<DocumentFragment>
  <button
    aria-label="Toggle theme"
    class="p-2 rounded focus:outline-none"
    data-testid="astra-theme-switch"
  >
    <svg
      aria-label="Apply dark theme"
      class="w-5 h-5"
      data-testid="astra-switcher-light-mode"
      fill="currentColor"
      role="image"
      viewBox="0 0 20 20"
    >
      <title
        id="dark-mode"
      >
        switch to dark mode
      </title>
      <path
        d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
